ClientSideEncryptionToCPKN/ClientSideKeyVaultKeyToCustomerManagedKey/samplesrc/Migration.cs

```diff
@@ -0,0 +1,130 @@
+         * 
+         * NOTE: This program requires the following to be stored in the App.Config file:
+         * Azure Active Directory Tenant ID - tenantId
+         * Service Principal Application ID - clientId
+         * Service Principal Password - clientSecret
+         * Azure Subscription ID - subscriptionId
+         * Resource Group Name - resourceGroup
+         * Storage Account Name - storageAccount
+         * Storage Account Connection String- connectionString
+         * Key Vault Key Uri - keyVaultKeyUri
+         * Key Wrap Algorithm - keyWrapAlgorithm
+         * 
+         * NOTE: This program uses names from Constants.cs, which should be edited as needed
+         * 
+         */
+        static void Main()
+        {
+            //Credentials of Service Principal
+            TokenCredential credential =
+                new ClientSecretCredential(
+                    Constants.tenantId,
+                    Constants.clientId,
+                    Constants.clientSecret
+                    );
+            
+            //File Path for local file used to download and reupload
+            string localPath = "./data" + Guid.NewGuid().ToString() + "/";
+            Directory.CreateDirectory(localPath);
+            string localFilePath = Path.Combine(localPath, Constants.fileName);
+
+            //Get Uri for Key Vault key
+            Uri keyVaultKeyUri = new Uri(Constants.keyVaultKeyUri);
+            //Create CryptographyClient using Key Vault Key
+            CryptographyClient cryptographyClient = new CryptographyClient(keyVaultKeyUri, credential);
+            //Set up Client Side Encryption Options used for Client Side Encryption
+            ClientSideEncryptionOptions clientSideOptions = new ClientSideEncryptionOptions(ClientSideEncryptionVersion.V1_0)
+            {
+                KeyEncryptionKey = cryptographyClient,
+                KeyWrapAlgorithm = Constants.keyWrapAlgorithm
+            };
+
+            //Create Blob Service Client
+            BlobServiceClient blobServiceClient = new BlobServiceClient(Constants.connectionString);
+
+            //Run Setup Function that creates and example container and blob
+            Setup.SetupForExample(
+                blobServiceClient,
+                Constants.containerName,
+                Constants.fileName,
+                Constants.encryptionScopeName,
+                clientSideOptions,
+                keyVaultKeyUri);
+
+            //Convert Client Side Encryption Blob to Server Side Encrytion with Customer Managed Keys
+            EncryptWithCustomerManagedKey(
+                Constants.connectionString,
+                Constants.containerName,
+                Constants.fileName,
+                localFilePath,
+                clientSideOptions,
+                Constants.encryptionScopeName);
+
+            //Delete downloaded files
+            Setup.CleanUp(localPath);
```

Comment:
i'd use try-finally to make sure clean up happens if errors are raised.

Btw. Cleanup should be part of the Migration rather than Setup.